tree/dataframe/test/CMakeLists.txt

```diff
@@ -43,6 +43,11 @@ if (imt)
    ROOT_ADD_GTEST(dataframe_concurrency dataframe_concurrency.cxx LIBRARIES ROOTDataFrame)
 endif()
 
+if (MSVC)
+  # TODO: remove this workaround for MS compiler bug #1441527 once fixed
+  string(REPLACE "-Od -Z7" "-O2" CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
+  string(REPLACE "-Z7" "" CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO}")
+endif()
```

Comment:
should the flags be restored to the original values afterwards? if not, should the flags be overridden at the project level?

Reply:
> should the flags be restored to the original values afterwards?

Well, yes, if it is possible. I have no experience. Let me try...

Reply:
You mean something like this:
```cmake
if (MSVC)
  # TODO: remove this workaround for MS compiler bug #1441527 once fixed
  string(REPLACE "-Od -Z7" "-O2" CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
  string(REPLACE "-Z7" "" CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO}")
endif()
ROOT_ADD_GTEST(datasource_more datasource_more.cxx LIBRARIES ROOTDataFrame)
if (MSVC)
  # restore the compiler flags
  string(REPLACE "-O2" "-Od -Z7" CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
  set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -Z7")
endif()
```

Reply:
I guess so :smile: what do you think?

Reply:
Doesn't work. Still working on it

Reply:
You should be able to use [`set_source_files_properties`](https://cmake.org/cmake/help/latest/command/set_source_files_properties.html)?

Reply:
@Axel-Naumann it doesn't work